src/data/weather.ts

```diff
@@ -538,3 +554,122 @@ export const getWeatherConvertibleUnits = (
   hass.callWS({
     type: "weather/convertible_units",
   });
+
+export const getForecast = (
+  weather_attributes?: WeatherEntityAttributes | undefined,
+  forecast_event?: ForecastEvent,
+  forecast_type?: ForecastType | undefined
+):
+  | {
+      forecast: ForecastAttribute[];
+      type: "daily" | "hourly" | "twice_daily";
+    }
+  | undefined => {
+  if (
+    forecast_type === "daily" &&
+    forecast_event?.type === "daily" &&
+    forecast_event?.forecast &&
+    forecast_event?.forecast?.length > 2
+  ) {
+    return { forecast: forecast_event.forecast, type: "daily" };
+  }
+  if (
+    forecast_type === "hourly" &&
+    forecast_event?.type === "hourly" &&
+    forecast_event?.forecast &&
+    forecast_event?.forecast?.length > 2
+  ) {
+    return { forecast: forecast_event.forecast, type: "hourly" };
+  }
+  if (
+    forecast_type === "twice_daily" &&
+    forecast_event?.type === "twice_daily" &&
+    forecast_event?.forecast &&
+    forecast_event?.forecast?.length > 2
+  ) {
+    return { forecast: forecast_event.forecast, type: "hourly" };
+  }
+  if (
+    forecast_type === "legacy" &&
+    weather_attributes?.forecast &&
+    weather_attributes.forecast.length > 2
+  ) {
+    const hourly = isForecastHourly(weather_attributes.forecast);
+    if (hourly === true) {
+      const dateFirst = new Date(weather_attributes.forecast![0].datetime);
+      const datelast = new Date(
+        weather_attributes.forecast![
+          weather_attributes.forecast!.length - 1
+        ].datetime
+      );
+      const dayDiff = datelast.getTime() - dateFirst.getTime();
+      const dayNight = dayDiff > DAY_IN_MILLISECONDS;
+      return {
+        forecast: weather_attributes.forecast,
+        type: dayNight ? "twice_daily" : "hourly",
+      };
+    }
+    return { forecast: weather_attributes.forecast, type: "daily" };
+  }
+  if (forecast_type === undefined) {
```

Comment:
Can't this be simplified by adding something like this at the top ? 

```ts
if (forecast_type === undefined) {
  forceast_type = forecast_event?.type || "legacy"
}

Reply:
Yes, indeed, I tried to simplify it 👍